cloud_functions/gcs_event_based_ingest/tests/test_gcs_ocn_bq_ingest_it.py

```diff
@@ -374,15 +357,54 @@ def test_external_query_IT(bq, gcs_data, gcs_external_config, dest_dataset,
         }
     }
     main.main(test_event, None)
-    sleep(3)    # Need to wait on async query job
-    validation_query_job = bq.query(f"""
-        SELECT
-            COUNT(*) as count
-        FROM
-          `{os.environ.get('GCP_PROJECT')}.{dest_dataset.dataset_id}.{dest_table.table_id}`
-    """)
-
     test_data_file = os.path.join(TEST_DIR, "resources", "test-data", "nation",
                                   "part-m-00001")
-    for row in validation_query_job.result():
-        assert row["count"] == sum(1 for _ in open(test_data_file))
+    expected_num_rows = sum(1 for _ in open(test_data_file))
+    try:
+        bq_wait_for_rows(bq, dest_table, expected_num_rows)
+    except TimeoutError as err:
+        raise AssertionError from err
+
+
+def bq_wait_for_rows(bq_client: bigquery.Client, table: bigquery.Table,
```

Comment:
Pivoting to this approach has two benefits:
1. table.get requests are free (v.s SELECT COUNT(*) queries which cost / consume slots)
1. speeds up tests ~50%  on average (depending on bq load slot pool latency).